xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-source/plugin.js

```diff
@@ -33,6 +33,8 @@
         // edited document is new (unsaved) because we want the converter to use the syntax specified by the template
         // rather than the default wiki syntax.
         sourceSyntax: editor.element.getAttribute('data-sourceDocumentSyntax'),
+        // Indicate if the content is supposed to be executed in a restricted context
+        wysiwygRestricted: editor.element.getAttribute('data-restricted') === 'true' ? 'true' : 'false',
```

Comment:
You can pass the boolean value and it will be properly serialized.

```suggestion
        wysiwygRestricted: editor.element.getAttribute('data-restricted') === 'true',
```

Reply:
Thanks, I was not sure so I did not take any chances :)